plugins/wasi_nn/wasinn_ggml.cpp

```diff
@@ -1162,22 +1253,31 @@ Expect<ErrNo> compute(WasiNNEnvironment &Env, uint32_t ContextId) noexcept {
                                                      CxtRef.LlavaImagePosition,
                                                  CxtRef.LlamaInputs.end());
     ReturnCode = evaluateTokens(GraphRef, GraphRef.LlamaContext,
-                                std::move(EmbdInputBeforeImage), NPast);
+                                std::move(EmbdInputBeforeImage), NPast, NPos);
     if (ReturnCode != ErrNo::Success) {
       spdlog::error(
           "[WASI-NN] GGML backend: failed to evaluate input tokens before image."sv);
       return ReturnCode;
     }
-    bool EvalImageStatus =
-        llava_eval_image_embed(GraphRef.LlamaContext, CxtRef.LlavaImageEmbd,
-                               static_cast<int>(GraphRef.BatchSize), &NPast);
+
+    bool EvalImageStatus = false;
+    if (GraphRef.Qwen2vl) {
```

Comment:
I am wondering if more vision models will come; should we use more booleans for the conditions, or should we have an `enum class` for it?

Reply:
Just add the `VisonModelType` for this.